### ISSUE TITLE
fingerprint module on PPProEE mobian

### ISSUE THREAD

**Issue by rosbeef** [egormanga/pinephone-fingerprint#5]
Can someone help me 
On Pinephone pro with mobian last kernel build 5.19.
I installed the fingerprint back cover.

I send you all the result of the command i understand that i could use to debug.

```
sudo i2cdetect -l
i2c-0	i2c       	rk3x-i2c                        	I2C adapter
i2c-1	i2c       	rk3x-i2c                        	I2C adapter
i2c-3	i2c       	rk3x-i2c                        	I2C adapter
i2c-4	i2c       	rk3x-i2c                        	I2C adapter
i2c-5	i2c       	rk3x-i2c                        	I2C adapter
**i2c-9	i2c       	i2c-4-mux (chan_id 0)           	I2C adapter**
```
^i think i2c-9 should be the pogo adapter as it it is more or less result as in your issu #3 

```
sudo i2cdetect -y 9
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:                         -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- UU -- -- -- 
20: -- -- UU -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- UU -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- --                         
```
all addresses seems to be in use
```
lsmod |grep i2c
inv_mpu6050_i2c        16384  0
inv_mpu6050            36864  3 inv_mpu6050_i2c,inv_mpu6050_spi
```
maybe by one of thoses modules

```
sudo i2cdetect -F 9
Functionalities implemented by /dev/i2c-9:
I2C                              yes
SMBus Quick Command              yes
SMBus Send Byte                  yes
SMBus Receive Byte               yes
SMBus Write Byte                 yes
SMBus Read Byte                  yes
SMBus Write Word                 yes
SMBus Read Word                  yes
SMBus Process Call               yes
SMBus Block Write                yes
SMBus Block Read                 no
SMBus Block Process Call         no
SMBus PEC                        yes
I2C Block Write                  yes
I2C Block Read                   yes

```

So the python give me a fail:
```
sudo python3 pia.py
Traceback (most recent call last):
  File "/home/mobian/Projects/mobian/fingerprint/pia.py", line 206, in <module>
    if (__name__ == '__main__'): exit(main())
  File "/home/mobian/Projects/mobian/fingerprint/pia.py", line 202, in main
    fp = PIA()
  File "/home/mobian/Projects/mobian/fingerprint/pia.py", line 20, in __init__
    self.open()
  File "/home/mobian/Projects/mobian/fingerprint/pia.py", line 36, in open
    fcntl.ioctl(self._fd, self._I2C_SLAVE, self.address)
OSError: [Errno 16] Device or resource busy

```
![Screenshot_20220810-221531](https://user-images.githubusercontent.com/29942937/184054263-6797d8ba-ac7d-4cb4-8c61-57ef9c35ab8b.png)

maybe i have to flash your firmware but (i by the FPback cover on 1may )

**Comment by egormanga** [comment]
The reader should always be at address `0x28`. Your `i2c-9` is — I'm pretty sure — NOT the pogo pin channel you're looking for.

**Comment by rosbeef** [comment]
Ok sorry for that. Based on that address should always be 0x28 i found that i2c-5 is the reader.
Then i modify the devpath to _i2c-5_ in the _pia.py_ script
and the result is somethinglike that :
```
< 55 aa 00 7f 00 00 00 00 da 2e
> 55 aa 00 ff 00 00 00 54 54 48 4d 33 36 36 32 23 35 31 30 2d 49 4e 54 23 46 50 36 23 41 56 41 4c 4f 4e 23 50 2d 31 32 23 53 50 49 23 32 30 32 31 30 34 32 33 2d 76 31 2e 30 30 7c 7c 45 4d 46 5f 65 74 35 78 78 20 31 2e 32 2e 33 2e 31 34 5f 6c 6f 67 7c 7c 31 2e 34 2e 33 2e 31 36 c3 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c 6c
bytearray(b'THM3662#510-INT#FP6#AVALON#P-12#SPI#20210423-v1.00||EMF_et5xx 1.2.3.14_log||1.4.3.16')
< 55 aa 00 05 00 00 00 00 83 75
> 55 aa 00 85 00 00 00 00 09 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20 20
bytearray(b'')
``` 
Is it working ? and if yes what is working ?

**Comment by egormanga** [comment]
Yeah, it is kinda working, just as on my device. We're currently stuck with the shitty firmware (which I came up with a semi-working workaround to), yet no answer from the author for like half a year now. I'll reiterate this project some time later in case the demand persists. Closing for now as resolved and unrelated further (we've got issue #2 already).
